src/app_selection.rs

```diff
@@ -0,0 +1,352 @@
+use std::ops::Deref;
+
+use holochain_client::AppInfo;
+use holochain_types::dna::DnaHash;
+use thiserror::Error;
+
+use crate::{config::AllowedAppIds, AdminCall};
+
+#[derive(Debug, PartialEq, Error)]
+pub enum AppSelectionError {
+    #[error("App is not installed on the conductor")]
+    NotInstalled,
+
+    #[error("App is not in the list of allowed apps")]
+    NotAllowed,
+
+    #[error("Multiple matching apps were found, could not determine which to call")]
+    MultipleMatching,
+}
+
+fn find_installed_app<'a>(
+    dna_hash: &DnaHash,
+    coordinator_identifier: &str,
+    installed_apps: &'a [AppInfo],
+) -> Result<&'a AppInfo, AppSelectionError> {
+    let mut found_apps = installed_apps.iter().filter(|a| {
+        // TODO: Use real `coordinator_identifier` when field available.
+        a.installed_app_id == coordinator_identifier
+            && a.manifest.app_roles().iter().any(|r| {
+                r.dna
+                    .installed_hash
```

Comment:
Is installed_hash expected to be present? I'd expect to have to look at the `cell_info` to get the DNA hashes in general